core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -1105,16 +1105,35 @@ class SparkContext(config: SparkConf) extends Logging with ExecutorAllocationCli
 
   /**
    * :: DeveloperApi ::
+   * Express a preference to the cluster manager for a given total number of executors. This will
+   * result in either canceling pending requests or filing additional requests.
+   * This is currently only supported in Yarn mode. Return whether the request is received.
+   */
+  @DeveloperApi
+  override def requestTotalExecutors(numExecutors: Int): Boolean = {
+    assert(master.contains("yarn") || dynamicAllocationTesting,
+      "Requesting executors is currently only supported in YARN mode")
+    schedulerBackend match {
+      case b: CoarseGrainedSchedulerBackend =>
+        b.requestTotalExecutors(numExecutors)
+      case _ =>
+        logWarning("Requesting executors is only supported in coarse-grained mode")
+        false
+    }
+  }
+
+  /**
+   * :: DeveloperApi ::
    * Request an additional number of executors from the cluster manager.
    * This is currently only supported in Yarn mode. Return whether the request is received.
    */
   @DeveloperApi
-  override def requestExecutors(numAdditionalExecutors: Int): Boolean = {
+  override def requestExecutors(numExecutors: Int): Boolean = {
```

Comment:
Changing this name is also an API break since in Scala you can refer to variables by name. I do agree `numExecutors` seems better since it's obvious that they are additional.